### ISSUE TITLE
Should any of these old modules be deprecated?

### ISSUE THREAD

**Issue by databyjp** [weaviate/weaviate#5790]
Weaviate's module system has become very rich over the last few years, and there are now over 30 modules. 

Given this, should we consider deprecating older modules such as these, especially where alternatives with more modern models exist?

- img2vec-neural
- qna-transformers
- qna-openai
- ref2vec-centroid
- sum-transformers
- text2vec-contextionary

Maybe except:
- ner-transformers
    - Perhaps not this, as [GLiNER model](https://huggingface.co/spaces/tomaarsen/gliner_medium-v2.1) is getting more and more attention in the AI space. We may add support for that model in the ner-transformers module.

**Comment by dirkkul** [comment]
> text2vec-contextionary

we heavily used this in our tests as a cheap vectorizer, where quality does not really matter

> ref2vec-centroid

what would the alternative be?

**Comment by daveatweaviate** [comment]
That could become an internal testing tool. Ollama also provides a low, or no, cost alternative

**Comment by databyjp** [comment]
> > ref2vec-centroid
> 
> what would the alternative be?

Yeah, I agree there's no obvious alternative to this. My questions was should we deprecate any/some, rather than let's deprecate all :).

**Comment by flipace** [comment]
I wonder - do you have any metrics about the different modules, to gauge usage and make decisions here backed on some data?
